pkg/policy/node.go

```diff
@@ -26,11 +26,16 @@ import (
 
 // Node to define hierarchy of rules
 type Node struct {
-	path      string
-	Name      string           `json:"name"`
-	Parent    *Node            `json:"-"`
-	Rules     []PolicyRule     `json:"rules,omitempty"`
-	Children  map[string]*Node `json:"children,omitempty"`
+	path     string
+	Name     string           `json:"name"`
+	Parent   *Node            `json:"-"`
+	Rules    []PolicyRule     `json:"rules,omitempty"`
+	Children map[string]*Node `json:"children,omitempty"`
+
+	// CoverAll marks policy nodes which should always have their
+	// rules applied regardless of whether the name/path of the node
+	// matches any of the labels in the search context.
+	CoverAll  bool `json:"coverAll,omitempty"`
```

Comment:
It took me a while to figure out the meaning of this.
IMO `CoverAll` means the node should cover **all** coverages in order to apply the rules.
This is the requirement for the kubernetes network policy labels [here](https://github.com/kubernetes/apimachinery/blob/9c89140a45325f83409e90359888133478207c7a/pkg/apis/meta/v1/types.go#L767)
```
"spec": {
    "podSelector": {
      "matchLabels": {
        "guestbook": "web",
        "k8s-app": "kubedns"
      }
    },
    "ingress": [
      {
        "from": [
          {
            "podSelector": {
              "matchLabels": {
                "io.cilium.reserved": "world"
              }
            }
          }
        ]
      }
    ]
  }
```
Will result into:
```
"guestbook-web": {
    "name": "guestbook-web",
    "rules": [
        {
            "coverage": [
                {
                    "key": "guestbook",
                    "value": "web",
                    "source": "k8s"
                },
                {
                    "key": "k8s-app",
                    "value": "kubedns",
                    "source": "k8s"
                }
            ],
            "allow": [
                {
                    "action": "always-accept",
                    "matchLabels": [
                        {
                            "key": "world",
                            "source": "reserved"
                        }
                    ]
                }
            ]
        }
    ],
    "coverAll": true
}
```